Framework/Algorithms/src/ResampleX.cpp

```diff
@@ -114,8 +114,23 @@ map<string, string> ResampleX::validateInputs() {
  */
 string determineXMinMax(MatrixWorkspace_sptr inputWS, vector<double> &xmins,
                         vector<double> &xmaxs) {
-  bool updateXMins = xmins.empty(); // they weren't set
-  bool updateXMaxs = xmaxs.empty(); // they weren't set
+  const size_t numSpectra = inputWS->getNumberHistograms();
+
+  // pad out the ranges by copying the first value to the rest that are needed
+  if (xmins.size() == 1) {
+    const double value = xmins.front();
+    while (xmins.size() < numSpectra)
+      xmins.push_back(value);
+  }
```

Comment:
Use std::vector::insert. The third form looks like what you've done here and below 

http://en.cppreference.com/w/cpp/container/vector/insert